mockns1/servehttp.go

```diff
@@ -0,0 +1,117 @@
+package mockns1
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// ServeHTTP is the request  handler for the mock service. This
+// should not be called directly in unit tests.
+func (s *Service) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	s.stopTimer()
+	defer s.startTimer()
+
+	if _, exists := s.tests[r.Method]; !exists {
+		notFoundResponse(w, "method")
+		return
+	}
+
+	tests, exists := s.tests[r.Method][r.RequestURI]
+	if !exists {
+		notFoundResponse(w, "uri")
+		return
+	}
+
+	defer r.Body.Close()
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte( // nolint: errcheck
+			fmt.Sprintf(`{"message": "unable to read request body: %s`, err),
+		))
+		return
+	}
+
+	var test *testCase
+	for _, t := range tests {
+		if !t.request.json {
+			if !assert.Equal(new(testifyT), t.request.body, body) {
+				continue
+			}
+		} else {
+			if !assert.JSONEq(new(testifyT), string(t.request.body), string(body)) {
+				continue
+			}
+		}
+
+		if !compareHeaders(t.request.headers, r.Header) {
+			continue
+		}
+
+		test = t
+		break
+	}
+
+	if test == nil {
+		notFoundResponse(w, "no test")
+		return
+	}
```

Comment:
Heh. This is what I get for trying to re-write code from memory rather than looking at my original code (Salesforce.com internal version that I've gotten permission to port to this project)...

I'm not a fan of long `if` statements like your example as I find them less readable, but I can definitely clean up `ServeHTTP()` by breaking things up better like my original code.

Please hold off on the merge, but I'll try to get this taken care of this weekend anyway. Thanks for pointing this out.

Reply:
@mburtless Here is my take on making that loop read easier. Have a look when you have a chance.

Reply:
Cool solution, that looks great.